client/app/account/user/emails/details/user-emails-details.controller.js

```diff
@@ -1,17 +1,39 @@
-angular.module("UserAccount").controller("UserAccountEmailsDetailsController", [
-    "$scope",
-    "$stateParams",
-    "UserAccount.services.emails",
-    "Alerter",
-    "$location",
-    "$translate",
+angular
+    .module("UserAccount")
+    .controller("UserAccountEmailsDetailsController", class UserAccountEmailsDetailsController {
+        constructor ($stateParams, $translate, Alerter, OvhApiMe) {
+            this.$stateParams = $stateParams;
+            this.$translate = $translate;
+            this.Alerter = Alerter;
+            this.OvhApiMe = OvhApiMe;
+        }
+
+        $onInit () {
+            this.emailNotification = null;
+            this.isLoading = false;
 
-    function ($scope, $stateParams, Emails, Alerter, $location, $translate) {
-        "use strict";
+            return this.getEmail();
+        }
 
-        $scope.previousPage = $location.search() && $location.search().previousPage ? $location.search().previousPage : 1;
+        getEmail () {
+            this.isLoading = true;
+            return this.OvhApiMe.Notification().Email().History().v6()
+                .get({ id: this.$stateParams.emailId }).$promise
+                .then((emailNotification) => {
+                    this.emailNotification = emailNotification;
+                    this.emailNotification.bodyCook = this.constructor.parseBodyIfThereIsSomethingToParse(this.emailNotification.body);
+
+                    // A toggle used to display raw or cooked body
+                    this.emailNotification.displayBodyCook = !!this.emailNotification.bodyCook;
+                    return this.emailNotification;
+                })
+                .catch((err) => this.Alerter.error(this.$translate.instant("otrs_email_detail_error", { t0: this.$stateParams.emailId }), err.data, "otrs_email_detail"))
+                .finally(() => {
+                    this.isLoading = false;
+                });
+        }
 
-        function parseBodyIfThereIsSomethingToParse (body) {
+        static parseBodyIfThereIsSomethingToParse (body) {
```

Comment:
Maybe we should rename this function to `parseBody` or `tryParseBody` ?

Reply:
Renamed to `parseBody` :)